### ISSUE TITLE
Nested syntax (map)

### ISSUE THREAD

**Issue by gkiely** [MananTank/eslint-plugin-validate-jsx-nesting#2]
Not sure if it's possible to detect, but currently this will go undetected.
```tsx
const List = () => {
  const items = [{ id: 1, title: 'hello'}, {id: 2, title: 'world'}];
  return (
    <p>
      {items.map((item) => (
        <h1 key={item.id}>{item.title}</h1>
      ))}
    </p>
  )
}
```

Screenshot:
<img width="395" alt="image" src="https://user-images.githubusercontent.com/1948935/179432008-d54f3e49-99cc-46f1-bf6e-3aab22901e9d.png">

**Comment by MananTank** [comment]
Yeah, This is one of those scenarios that is either impossible or at the very least - extremely hard to detect with just static code analysis. 

To detect such types of issues we would need to run the code

<br/>

Other Examples where static analysis will not be able to detect invalid nestings: 

<br/>

```jsx
<p> {foo} </p>
```

because we can not know what `foo` is without running the code

<br/>

```jsx
<p> <Foo /> </p>
```

because we can not know what `<Foo />` returns without running the code.

<br/>

I should probably include the limitations of this plugin in the README to make it clear.

---

Technically it is possible to detect invalid nesting in the example you've shown, but that are so many edge cases that it's just not worth looking inside `JSXExpressionContainer`s, and it may even end up showing false positives. 

It's best to leave validation of Dynamic structures up to the runtime validator and validate only static structures via the ESLint plugin.

**Comment by gkiely** [comment]
Good to know thank you for the detailed explanation!
